pipelines/controller-cli/src/main.py

```diff
@@ -0,0 +1,245 @@
+        return f"Error deleting snapshot: {e}"
+
+
+def dwh(args) -> str:
+    try:
+        if hasattr(args, "snapshot_id"):
+            response = delete_snapshot(args)
+        else:
+            response = _make_api_request(HTTP_GET, f"{args.url}/dwh")
+        process_response(response, args)
+    except requests.exceptions.RequestException as e:
+        print(f"Error processing: {e}")
+
+
+def main():
+    logging.basicConfig(level=logging.INFO)
+    parser = argparse.ArgumentParser(
+        description="The CLI tool for fhir-data-pipes",
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+    parser.add_argument(
+        "url",
+        type=str,
+        help="url of the pipeline controller's REST API",
+    )
+    subparsers = parser.add_subparsers(
+        dest="command",
+        help=f'{", ".join(str(value) for value in COMMAND_MAP.values())} '
+        "are the available commands.",
+        required=True,
+    )
+
+    config_parser = subparsers.add_parser("config", help="show config values")
+    config_parser.add_argument(
+        "--config-name", "-cn", required=False, help="name of the configuration key"
+    )
+    config_parser.set_defaults(func=config)
+
+    next_parser = subparsers.add_parser("next", help="show the next scheduled run")
+    next_parser.set_defaults(func=next)
+
+    status_parser = subparsers.add_parser(
+        "status", help="show the status of the pipeline"
+    )
+    status_parser.set_defaults(func=status)
+
+    run_modes = ", ".join(mode.lower() for mode in RUN_MODES)
+    run_parser = subparsers.add_parser("run", help="run the pipeline")
+    run_parser.add_argument(
+        "--mode",
+        "-m",
+        choices=([mode.lower() for mode in RUN_MODES]),
+        required=True,
+        help=f"the runType argument, options are {run_modes}",
+    )
+    run_parser.set_defaults(func=run)
+
+    tables_parser = subparsers.add_parser("tables", help="create resource tables")
+    tables_parser.set_defaults(func=tables)
+
+    logs_parser = subparsers.add_parser("logs", help="show logs")
+    logs_parser.add_argument("--download", action="store_true")
+    logs_parser.add_argument(
+        "--filename",
+        "-f",
+        required=False,
+        help="name of the downloaded file, default error.log",
+    )
+    logs_parser.set_defaults(func=logs)
+
+    dwh_parser = subparsers.add_parser("dwh", help="show a list of dwh snapshots")
+    dwh_parser.set_defaults(func=dwh)
+    dwh_sub_parsers = dwh_parser.add_subparsers(dest="subcommand")
+    dwh_delete_parser = dwh_sub_parsers.add_parser(
+        "delete",
+        help="deletes a snapshot given an id. use 'controller {url} dwh' "
+        "to get list of snapshot ids",
+    )
+    dwh_delete_parser.add_argument(
+        "--snapshot-id",
+        "-si",
+        required=True,
+        help=f"the id of the snapshot in the format "
+        "dwh/controller_DEV_DWH_TIMESTAMP_2025_08_14T17_47_15_357080Z",
```

Comment:
Please either expand this `help` or make it more generic to show its relationship to [`dwhRootPrefix`](https://github.com/google/fhir-data-pipes/blob/58d1af530390cea080231743f663454126e73b6b/pipelines/controller/src/main/java/com/google/fhir/analytics/DataProperties.java#L72).